features/steps/givens/attributes.py

```diff
@@ -25,6 +25,19 @@ def step_impl(context, attribute, value):
     )
 
 
+@given("Its attribute {attribute} {condition} with {prefix}")
+def step_impl(context, attribute, condition, prefix):
+    assert condition in ('starts', 'does not start')
+
+    if condition == 'starts':
+        context.instances = list(
+            filter(lambda inst: hasattr(inst, attribute) and str(getattr(inst, attribute)).startswith(prefix), context.instances)
```

Comment:
```suggestion
            filter(lambda inst: hasattr(inst, attribute) and str(getattr(inst, attribute, '')).startswith(prefix), context.instances)
```

I would probably add a default to the getattr just in order to not have it throw an exception in case of a non-existing attr. That would be our mistake in the scenario writing though, but still.